plugins/inputs/internet_speed/internet_speed.go

```diff
@@ -65,6 +65,7 @@ func (is *InternetSpeed) Gather(acc telegraf.Accumulator) error {
 	fields["download"] = s.DLSpeed
 	fields["upload"] = s.ULSpeed
 	fields["latency"] = timeDurationMillisecondToFloat64(s.Latency)
+	fields["provider"] = s.Sponsor
```

Comment:
Can you please make this a tag as it might be helpful to search/filter for it and it is string anyway...

Reply:
Hmm, I’m not sure about that. 🤔

Reply:
@Hipska, do you care to outline why you are not sure and about what? ;-)

Reply:
If it needs to be a tag.

The selected provider can be different every time, so it is more like a metric field to me.